engine/apps/twilioapp/models/twilio_sender.py

```diff
@@ -0,0 +1,37 @@
+from django.db import models
+from polymorphic.models import PolymorphicModel
+from twilio.rest import Client
+
+
+class TwilioAccount(models.Model):
+    name = models.CharField(max_length=100)
+    account_sid = models.CharField(max_length=64, null=False, blank=False, unique=True)
+    auth_token = models.CharField(max_length=64, null=True, default=None)
+    api_key_sid = models.CharField(max_length=64, null=True, default=None)
+    api_key_secret = models.CharField(max_length=64, null=True, default=None)
+
+    def get_twilio_api_client(self):
+        if self.api_key_sid and self.api_key_secret:
+            return Client(self.api_key_sid, self.api_key_secret, self.account_sid)
+        else:
+            return Client(self.account_sid, self.auth_token)
+
+
+class TwilioSender(PolymorphicModel):
```

Comment:
Maybe it would be easier to have this model as an abstract Django model? Having an abstract model here will keep things DRY without having to deal with any of the polymorphic stuff (which sometimes can make migrations more complex and introduce [random weirdness](https://github.com/grafana/oncall/pull/1652)).